SU-HSDO/suhumsci
# feat:stn-564-dark-refactor-masthead-setting

# READY FOR REVIEW

## Summary
There were two goals for this card:
1. Add a new theme setting to switch from a light theme to a dark theme in the masthead. Which should work similarly to what was done in the local footer
2. Refactor the existing footer class for a modifier class to be a reusable utility class that will eventually be able to be used in multiple places.
- All instances of the hb-local-footer--dark class should be renamed to hb-dark-pattern
- All CSS for the hb-dark-pattern should be moved to a single partial in the utilities section of the CSS.
- ❗ Any CSS that in a mixin should not be moved. The amount of effort to refactor these mixins is out of scope of this card.

Jira Card:[ STN-564](https://sparkbox.atlassian.net/browse/STN-564)

## Need Review By (Date)
10/2/20

## Urgency
medium

## Steps to Test
1. `npm run test`
2. Clear Drupal Cache 
3. Set your theme to Colorful (Appearance > Settings)
4. To test the masthead dark mode setting, you'll want to turn it on here (Appearance > Settings > Humsci Colorful)
![image](https://user-images.githubusercontent.com/9355698/94852629-2f950480-03f8-11eb-98e2-fc4a504a78cd.png)
5. After you set the masthead to dark it should look like the image below and you should see `hb-dark-pattern` in the header upon inspection.
![image](https://user-images.githubusercontent.com/9355698/94822076-46bffc00-03d0-11eb-984e-362f24c3b6b8.png)
6. Repeat those steps for the Traditional theme and check the other color pairings for those themes if you'd like.
7. We also refactored the css into a utility class, `hb-dark-pattern` if you apply this class on other elements the dark made background and default white text inversion should be applied. This will be polished and more widely applied as we add components to dark mode options in future cards.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
